### ISSUE TITLE
Set defaults on a per class basis

### ISSUE THREAD

**Issue by sorenmalling** [wbsply/route-annotation#1]
How about we introduce a `Defaults` annotation kind of like this

```
#[Route\Defaults(prefix: 'uri-prefix', format: 'json')
```

**Comment by mhsdesign** [comment]
so like:

```
#[Route\Defaults(prefix: 'uri-prefix', format: 'json')
class MyController
{
   #[Route('lol', method: "GET")
   public function lolAction()
   {}
}
```

would result in `uri-prefix/lol` ?

**Comment by bwaidelich** [comment]
Why not re-use the existing names from the routing config (it's not super nice, but at least that would make it consistent), i.e.:
```php
class MyController
{
   #[Route(uriPatter: "foo", defaults: ['@format': 'json'], httpMethods: ['GET'])
   public function lolAction()
   {}
}

**Comment by sorenmalling** [comment]
> Why not re-use the existing names from the routing config (it's not super nice, but at least that would make it consistent), i.e.:

My attempt to make config key more simple. `format` is a part of `default` - but we read all other `defaults` from the position of the annotation. So, `defaults: {@format: ..}` can simply be a `format` property :)

**Comment by sorenmalling** [comment]
> i dont really understand but like:
> 
> ```
> #[Route\Defaults(prefix: 'uri-prefix', format: 'json')
> class MyController
> {
>    #[Route('lol', method: "GET")
>    public function lolAction()
>    {}
> }
> ```
> 
> would result in `uri-prefix/lol` ?

Exactly, routes for this class all have `contacts` as a "prefix" - son instead of repeating yourself in each annotation, we can "prefix" the class. It's possible now by setting a `Route('prefix')` on the class - this is about making  single purpose annotations

**Comment by mhsdesign** [comment]
yes id consider it more clean to have a separate annotation for this than to reuse the [existing one](https://github.com/wbsply/route-annotation#example) though i dont like your proposed name ^^ naming things i know ...

**Comment by mficzel** [comment]
I would like to see the first implementation of annotation routes be close to the yaml configuration and not introduce other new features yet. 

We may want that later after actually working with route annotations. If it can be treated as a seperate feature it probably should.
